pytket/binders/passes.cpp

```diff
@@ -264,24 +264,26 @@ PYBIND11_MODULE(passes, m) {
       .def("__repr__", &BasePass::to_string)
       .def(
           "to_dict",
-          [](const BasePass &base_pass) {
-            return py::object(base_pass.get_config()).cast<py::dict>();
-          },
+          [](const BasePass &base_pass) { return serialise(base_pass); },
           ":return: A JSON serializable dictionary representation of the Pass.")
       .def_static(
           "from_dict",
-          [](const py::dict &base_pass_dict) {
-            return json(base_pass_dict).get<PassPtr>();
+          [](const py::dict &base_pass_dict,
+             const std::map<
+                 std::string, std::function<Circuit(const Circuit &)>>
+                 &custom_deserialisation) {
```

Comment:
This argument should be optional, then we don't break the API.

Reply:
done